BloomFilter.md

```diff
@@ -47,23 +47,33 @@ pushdown even in columns with high cardinality and when space is at a premium.
 The initial Bloom filter algorithm in Parquet is implemented using a combination of two
 Bloom filter techniques.
 
-First, the block Bloom filter algorithm from Putze et al.'s "Cache-, Hash- and
-Space-Efficient Bloom filters"[2] is used. This divides a filter into many tiny Bloom
-filters, each one of which is called a "block". In Parquet's initial implementation, each
-block is 256 bits. When inserting or finding a value, part of the hash of that value is
-used to index into the array of blocks and pick a single one. This single block is then
-used for the remaining part of the operation.
-
-Second, within each block, this implementation uses the folklore split Bloom filter
-technique, as described in section 2.1 of "Network Applications of Bloom Filters: A
-Survey"[5]. This divides the 256 bits in each block up into eight contiguous 32-bit lanes
-and sets or checks one bit in each lane.
+First, the block Bloom filter algorithm from Putze et al.'s [Cache-, Hash- and Space-Efficient
+Bloom filters](http://algo2.iti.kit.edu/documents/cacheefficientbloomfilters-jea.pdf) is used.
+The block Bloom filter consists of a sequence of small Bloom filters, each of which can fit
+into one cache-line. For best performance, those small Bloom filters are loaded into memory
+cache-line-aligned. For each potential element, the first hash value selects the Bloom filter block
+to be used. Additional hash values are then used to set or test bits as usual, but only inside
+this one block. As for Parquet's initial implementation, each block is 256 bits. When inserting or
+finding value, the first hash of that value is used to index into the array of blocks and pick a
+single one. This single block is then used for the remaining part of the operation.
+
+Second, the remaining part of the operation within the single block uses the folklore split Bloom
```

Comment:
I have read the linked material and found a description of this technique, but it was not referred to as "folklore split" Bloom filter technique.

Reply:
While Broder and Mitzenmacher did not call it "folklore" in the linked paper, they say "sometimes Bloom filters are described slightly differently". It is common in academic writing for CS techniques without a known inventor to be attributed to folklore.

Reply:
As I wrote I found this technique in the end, I was just a little confused because from this wording I expected it to be referred to as folklore split Bloom filter technique in the paper as well. Not a big issue, I just mentioned it.